src/rust/inetstack/protocols/layer4/udp/peer.rs

```diff
@@ -155,21 +154,11 @@ impl<N: NetworkRuntime> SharedUdpPeer<N> {
     }
 
     /// Consumes the payload from a buffer.
-    pub fn receive(&mut self, ipv4_hdr: Ipv4Header, buf: DemiBuffer) {
+    pub fn receive(&mut self, src_ipv4_addr: Ipv4Addr, dest_ipv4_addr: Ipv4Addr, hdr: UdpHeader, buf: DemiBuffer) {
         timer!("udp::receive");
-        // Parse datagram.
-        let (hdr, data): (UdpHeader, DemiBuffer) = match UdpHeader::parse(&ipv4_hdr, buf, self.checksum_offload) {
-            Ok(result) => result,
-            Err(e) => {
-                let cause: String = format!("dropping packet: unable to parse UDP header");
-                warn!("{}: {:?}", cause, e);
-                return;
-            },
-        };
-        debug!("UDP received {:?}", hdr);
 
-        let local: SocketAddrV4 = SocketAddrV4::new(ipv4_hdr.get_dest_addr(), hdr.dest_port());
-        let remote: SocketAddrV4 = SocketAddrV4::new(ipv4_hdr.get_src_addr(), hdr.src_port());
+        let local: SocketAddrV4 = SocketAddrV4::new(dest_ipv4_addr, hdr.dest_port());
+        let remote: SocketAddrV4 = SocketAddrV4::new(src_ipv4_addr, hdr.src_port());
```

Comment:
Could we remove from this PR any changes that do not concern source tree refactoring?